core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveEmptyUpdate.java

```diff
@@ -1,72 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package io.trino.sql.planner.iterative.rule;
-
-import com.google.common.collect.ImmutableList;
-import io.trino.matching.Captures;
-import io.trino.matching.Pattern;
-import io.trino.sql.planner.iterative.Rule;
-import io.trino.sql.planner.plan.TableFinishNode;
-import io.trino.sql.planner.plan.ValuesNode;
-import io.trino.sql.tree.GenericLiteral;
-import io.trino.sql.tree.Row;
-
-import static io.trino.sql.planner.plan.Patterns.emptyValues;
-import static io.trino.sql.planner.plan.Patterns.exchange;
-import static io.trino.sql.planner.plan.Patterns.source;
-import static io.trino.sql.planner.plan.Patterns.tableFinish;
-import static io.trino.sql.planner.plan.Patterns.update;
-
-/**
- * If the predicate for an update is optimized to false, the target table
- * scan of the update will be replaced with an empty values node.
- * This type of plan cannot be executed and is meaningless anyway, so we
- * replace the entire plan node with a `values` node.
- * <p>
- * Transforms
- * <pre>
- *  - TableFinish
- *    - Exchange
- *      - Update
- *        - empty Values
- * </pre>
- * into
- * <pre>
- *  - Values (0)
- * </pre>
- */
-public class RemoveEmptyUpdate
-        implements Rule<TableFinishNode>
-{
-    private static final Pattern<TableFinishNode> PATTERN = tableFinish()
-            .with(source().matching(exchange()
```

Comment:
> Remove empty UPDATE also when wrapped with exchange 

i think the "wrapped with exchange" case was satisfied.

Reply:
I merged https://github.com/trinodb/trino/pull/13755 without resolving https://github.com/trinodb/trino/pull/13755#discussion_r951808946 because I don't know whether it matters. I didn't want to have a dead code.